src/watertap_contrib/reflo/costing/units/multi_effect_crystallizer.py

```diff
@@ -0,0 +1,381 @@
+        costing_package.multi_effect_crystallizer.heat_exchanger_endplates_capital_factor
+        * (dimensionless_hx_area)
+        ** costing_package.multi_effect_crystallizer.heat_exchanger_endplates_capital_exponent,
+        to_units=costing_package.base_currency,
+    )
+
+    capital_cost_hx_effect += capital_cost_hx_endplates
+
+    return capital_cost_hx_effect
+
+
+def cost_crystallizer_effect_by_crystal_mass(effect):
+    """
+    Mass-based capital cost for forced circulation crystallizer
+    """
+
+    capital_cost_effect = pyo.units.convert(
+        (
+            costing_package.multi_effect_crystallizer.iec_percent
+            * costing_package.multi_effect_crystallizer.fob_unit_cost
+            * (
+                sum(
+                    effect.properties_solids[0].flow_mass_phase_comp["Sol", j]
+                    for j in effect.config.property_package.solute_set
+                )
+                / costing_package.multi_effect_crystallizer.ref_capacity
+            )
+            ** costing_package.multi_effect_crystallizer.ref_exponent
+        ),
+        to_units=costing_package.base_currency,
+    )
+
+    return capital_cost_effect
+
+
+def cost_crystallizer_effect_by_volume(effect):
+    """
+    Volume-based capital cost for forced circulation crystallizer
+    """
+
+    capital_cost_effect = pyo.units.convert(
+        (
+            costing_package.multi_effect_crystallizer.volume_cost
+            * (
+                (
+                    pyo.units.convert(
+                        effect.volume_suspension
+                        * (effect.height_crystallizer / effect.height_slurry),
+                        to_units=(pyo.units.ft) ** 3,
+                    )
+                )
+                / pyo.units.ft**3
+            )
+            ** costing_package.multi_effect_crystallizer.vol_basis_exponent
+        ),
+        to_units=costing_package.base_currency,
+    )
+
+    return capital_cost_effect
+
+
+def _cost_effect_flows(effect, effect_number):
+
+    costing_package.cost_flow(
+        pyo.units.convert(
+            (
+                effect.magma_circulation_flow_vol
+                * effect.dens_mass_slurry
+                * Constants.acceleration_gravity
+                * effect.pump_head_height
+                / effect.efficiency_pump
+            ),
+            to_units=pyo.units.kW,
+        ),
+        "electricity",
+    )
+
+    costing_package.cost_flow(
+        effect.properties_solids[0].flow_mass_phase_comp["Sol", "NaCl"],
+        "NaCl_recovered",
+    )
+
+    if effect_number == 1:
+        effect.steam_flow = pyo.units.convert(
+            (effect.work_mechanical[0] / _compute_steam_specific_energy(effect)),
+            to_units=pyo.units.m**3 / pyo.units.s,
+        )
+        costing_package.cost_flow(
+            effect.steam_flow,
+            "steam",
+        )
+        # costing_package.cost_flow(effect.heating_steam[0].flow_vol_phase["Vap"],
```

Comment:
@Zhuoran29 since we added the ```heating_steam``` state block, would it be correct to just cost the ```flow_vol_phase["Vap"]``` here?

Reply:
done in 045b9ab

Reply:
It'd be correct. The functions in `_compute_steam_specific_energy` below is the same as in the water property package.

Reply:
Ok- I believe it will change the costing results. The crystallizer prop pack doesn't have the specific volume property, but if the property model is calculating it in a different way then I suppose that is just as good. 

So in theory the ```steam_flow``` calculated here should be equal to ```flow_vol_phase["Vap"]```?

Reply:
believe we figured this out.